File: PythonCode/isPrime.py
# 判断1-n的质数
# def isPrime(result, n):
#     for i in range(2, n):
#         flag = True
#         for j in range (2, i):
#             if i % j == 0:
#                 flag = False
#                 break
#         if flag:
#             result.append(i)
#     print(result)

def isPrime(n):
    if n <= 1:
        return False
    for i in range(2, n):
        if n % i == 0:
            return False
    return True


if __name__ == '__main__':
    result = []
    # isPrime(result, 100)
    for i in range(1, 101):
        if isPrime(i):
            result.append(i)
    print("Prime number from 1 to 100 are:\n", result)

File: C_code/isPrime.c
/**********输出1-100之间的质数*/
#include <stdio.h>
#include <stdbool.h>

bool is_prime(int num) {
    if (num <= 1) {
        return false;
    }
    for (int i = 2; i * i <= num; ++i) {
        if (num % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    printf("Prime numbers from 1 to 100 are:\n");
    for (int i = 1; i <= 100; ++i) {
        if (is_prime(i)) {
            printf("%d ", i);
        }
    }
    printf("\n");
    return 0;
}
